mundane-bytes/src/main/scala/com/ambiata/mundane/bytes/Buffer.scala

```diff
@@ -65,17 +65,18 @@ object Buffer {
     wrapArray(b.bytes, b.offset + o, b.length - o)
   }
 
-  def reset(b: Buffer): Unit =
+  def reset(b: Buffer): Buffer = {
     b._length = 0
+    b
+  }
 
   /** Ensure there is enough space in the array, and if not return a _new_ [[Buffer]] and array with the original values */
   def allocate(b1: Buffer, i: Int): Buffer =
-    if (b1.length + i <= b1.bytes.length) {
+    if (b1.length + b1.offset + i <= b1.bytes.length) {
```

Comment:
Still don't know if it this is correct, I think I may of glossed over the sematics of this when I read it last time - but I expected `i` to be a new length, as in you are ensuring that the buffer is big enough for `offset + length`.